cpp/ql/lib/semmle/code/cpp/models/implementations/Printf.qll

```diff
@@ -50,6 +50,8 @@ private class Fprintf extends FormattingFunction, NonThrowingFunction {
   override int getFormatParameterIndex() { result = 1 }
 
   override int getOutputParameterIndex(boolean isStream) { result = 0 and isStream = true }
+
+  override int getFirstFormatArgumentIndex() { result = 2 }
```

Comment:
I don't fully understand this, but it's what's needed to remove false-positives on the `asterisk` project. The declaration of `int fprintf()` in the last commit (https://github.com/github/codeql/pull/17775/commits/5315a5cfbf34641addabffa777e662bd147fce6a) replicates what happened in the database. I briefly forgot why I needed that line.

This line is safe because class `Fprintf` always has 2 arguments.

Reply:
The root cause is actually that the extractor sometimes fails to mark function declarations as implicit. Hmm. Maybe that's the fix.

Reply:
Fixed instead using https://github.com/github/codeql/pull/17775/commits/4341fab79472324ef6e39e09329cb563a3e194c8. I couldn't find a nice way to identify these extraction errors but hopefully the solution makes sense.